source/jormungandr/jormungandr/realtime_schedule/timeo.py

```diff
@@ -129,7 +133,7 @@ def _get_passages(self, timeo_resp):
         for next_expected_st in next_st.get('NextExpectedStopTime', []):
             # for the moment we handle only the NextStop and the direction
             dt = self._get_dt(next_expected_st['NextStop'])
-            direction = next_expected_st.get('Destination')
+            direction = self._get_direction_name(next_expected_st.get('Terminus'), next_expected_st.get('Destination'))
```

Comment:
you're sure this works ?

why do you need the `Terminus` and the `Destination` ?